src/core/server/saved_objects/service/lib/filter_utils.test.ts

```diff
@@ -494,5 +494,33 @@ describe('Filter Utils', () => {
         },
       ]);
     });
+
+    test('Validate multiple items nested filter query through KueryNode', () => {
+      const validationObject = validateFilterKueryNode({
+        astFilter: esKuery.fromKueryExpression(
+          'alert.attributes.actions:{ actionTypeId: ".server-log" AND actionRef: "foo" }'
+        ),
+        types: ['alert'],
+        indexMapping: mockMappings,
+      });
+
+      // nodes will have errors in the array
+      expect(validationObject).toEqual([
+        {
+          astPath: 'arguments.1.arguments.0',
+          error: null,
+          isSavedObjectAttr: false,
+          key: 'alert.attributes.actions.actionTypeId',
+          type: 'alert',
+        },
+        {
+          astPath: 'arguments.1.arguments.1',
+          error: null,
+          isSavedObjectAttr: false,
+          key: 'alert.attributes.actions.actionRef',
+          type: 'alert',
+        },
+      ]);
+    });
```

Comment:
make sense, I will try to add it

Reply:
I added a jest integration test for the saved objects repository in https://github.com/elastic/kibana/pull/84326 this kind of test would be nice to add there instead of an `api_integration` test. Don't want to block you on that PR, so you could just copy the test file to your branch.